src/proposer/proposer.cpp

```diff
@@ -29,15 +30,83 @@
 
 namespace proposer {
 
-class ProposerStub : public Proposer {
+bool GenerateBlock(staking::ActiveChain &active_chain,
+                   staking::TransactionPicker &transaction_picker,
+                   proposer::BlockBuilder &block_builder,
+                   proposer::Logic &logic,
+                   CWallet *wallet,
```

Comment:
According to the implementation, wallet couldn't be nullptr.

Reply:
I'll make it a reference.